### PR TITLE
let data to be deleted by user or accumulo ageoff filter

### LINKED ISSUE #26: To delete old data in Accumulo
COMET never really deletes data in Accumulo. 
Need a mechanism to delete aged and unused data.

### PR DESCRIPTION
**Pull request by efu39**
To avoid data nodes out of disk space eventually, Comet will delete data if requested by user and also let data ageoff if there is no access activity.

Changes in this pull request:

1. When user requests DeleteScope, instead of fake deletion (only mark ifDeleted and update in the Accumulo), now Comet will perform actual deletion to the Accumulo. 

2. Turn on Accumulo AgeOffFilter, and perform internal write to update timestamp if the data is too old during read or enumerate operation. A new environment variable "COMET_RECORD_EXPIRE_TIME" is introduced, records are expected to access (i.e. Read/Enumerate/Write) at least once in this period of time in order to keep records alive.

3. COMET still keeps the original user write timestamp in case this information is needed in future.
   Previous format in Accumulo record value: 
**{ifDeleted, writeToken, scopeValue, Comet_version, deletionTimeStamp }**
   Now New format in Accumulo record value : 
**{ifDeleted, writeToken, scopeValue, Comet_version, deletionTimeStamp, userWriteTimeStamp }**
   Note: Althouth **ifDeleted** and **deletionTimeStamp** are not used anymore, keep it to be backward compatible with previous versions.

4. test folder
    docker-compose-test.yml: to setup an test instance on comet-hn2 using port 8222
    test_comet.py          : testcases can run by pytest

### CODE REVIEW

File: comet-accumulo/src/main/java/org/renci/comet/CometInitializer.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 @EnableSwagger2
 @ComponentScan(basePackages = { "org.renci.comet", "org.renci.comet.api" })
 public class CometInitializer implements CommandLineRunner {
-	public static final String COMET_VERSION = "1.0.0";
+	public static final String COMET_VERSION = "1.0.5";
 	public static final String COMET_API_VERSION = "1.0.0";
 	public static final String COMET_SCHEMA_VERSION = "1.0.0";
```
Comment (kthare10) [inline comet-accumulo/src/main/java/org/renci/comet/CometInitializer.java]: Schema version should be increased too, since we are adding new fields.

Comment (efu39) [reply comet-accumulo/src/main/java/org/renci/comet/CometInitializer.java]: good catch, thanks!

Comment (efu39) [reply comet-accumulo/src/main/java/org/renci/comet/CometInitializer.java]: Just pushed a new commit to update this

**Review by kthare10 — APPROVED** [review]

**Comment by kthare10** [comment]
looks good!

**Pull request merged**